File: src/my_cpp_pkg/src/hw_status_publisher.cpp
#include "rclcpp/rclcpp.hpp"
#include "my_robot_interfaces/msg/hardware_status.hpp"


class HwStatusPublisher: public rclcpp::Node
{
    public:

        HwStatusPublisher(): Node("hw_status_publisher")
        {
            publisher_ = this->create_publisher<my_robot_interfaces::msg::HardwareStatus>("hw_status",10);
            timer_ = this->create_wall_timer(std::chrono::seconds(1),std::bind(&HwStatusPublisher::callback_publish_hw_status,this));
            RCLCPP_INFO(this->get_logger(),"Hardware Status publisher node has been started.");
        }

    private:

        void callback_publish_hw_status()
        {
            auto msg = my_robot_interfaces::msg::HardwareStatus();
            msg.temperature  = 45;
            msg.are_motors_ready = true;
            msg.debug_message  = "Set to ttrue";
            publisher_->publish(msg);
        }

        rclcpp::Publisher<my_robot_interfaces::msg::HardwareStatus>::SharedPtr publisher_;

        rclcpp::TimerBase::SharedPtr timer_;

};

int main(int argc,char** argv)
{
    rclcpp::init(argc,argv);
    auto node = std::make_shared<HwStatusPublisher>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}

File: src/my_py_pkg/my_py_pkg/hw_status_publisher.py
#!/usr/bin/env python3


import rclpy
from rclpy.node import Node

from my_robot_interfaces.msg import HardwareStatus
#from my_robot_interfaces.msg import HardwareStatus


class HwStatusPublisherNode(Node):
    def __init__(self):
        super().__init__("Hardware_status_publisher")
        self.hw_publisher = self.create_publisher(HardwareStatus,"hw_status",10)

        self.timer_ = self.create_timer(0.5,self.callback_publish_hw)
        
        self.get_logger().info("Hardware Status Publisher has been started.")


    def callback_publish_hw(self):

        msg = HardwareStatus()

        msg.temperature = 45
        msg.are_motors_ready = False
        msg.debug_message = "Set AS"

        self.hw_publisher.publish(msg)


        




def main(args=None):
    rclpy.init(args=args)
    node = HwStatusPublisherNode()
    rclpy.spin(node)
    rclpy.shutdown()


if __name__ == "__main__":
    main()
